packages/playwright-core/src/server/injected/roleUtils.ts

```diff
@@ -461,6 +461,60 @@ export function getElementAccessibleDescription(element: Element, includeHidden:
   return accessibleDescription;
 }
 
+// https://html.spec.whatwg.org/multipage/input.html#the-input-element
+export const kInputValidityAttributes = ['required', 'pattern', 'min', 'max', 'minlength', 'maxlength', 'step', 'type', 'value'];
+
+export function hasValidationAttributes(element: Element): boolean {
+  if (!(element instanceof HTMLInputElement))
```

Comment:
`validity` is defined for a lot of elements, e.g. `<select>`, `<textarea>`, `<object>`, `<button>`, `<fieldset>`, probably more. I'd drop this check, and assume that `element.validity?.valid === false` means it is invalid.